g2p/mappings/langs/network_lite.py

```diff
@@ -0,0 +1,164 @@
+from collections import deque
+from typing import Any, Iterable, Iterator
+
+
+class DiGraph:
+    """A simple directed graph class
+
+    Most functions raise KeyError if called with a node u or v not in the graph.
+    """
+
+    def __init__(self) -> None:
+        """Contructor, empty if no data, else load from data"""
+        self._edges: dict = {}
+
+    def clear(self):
+        """Clear the graph"""
+        self._edges.clear()
+
+    def update(self, edges: Iterable, nodes: Iterable):
+        """Update the graph with new edges and nodes"""
+        for node in nodes:
+            self.add_node(node)
+        for u, v in edges:
+            self.add_edge(u, v)
+
+    def add_node(self, u):
```

Comment:
Should be `u: Hashable` as not all types can actually be keys in a dict (likewise for `add_edge`, etc)